docs/aws/provisioner-launch-templates-options.md

```diff
@@ -0,0 +1,213 @@
+wanted to support both ARM and x86 in the same cluster, they'd have to
+create two separate provisioners.
+
+The provisioner does not allow the customer to specify a
+`kubernetes.io/arch` label in the `labels` section, so there is no
+risk of a conflict between that and the launch template.
+
+Another problem users might run into is that their pod spec might
+specify a node selector for an architecture that the provisioner
+doesn't support.
+
+# Solutions
+
+## Label Naming
+
+It seems like we should standardize on a label naming convention, and
+since ACK is using it, and it's short, we should use `k8s.aws` as the
+root suffix going forward. (Obviously while maintaining
+backwards-compatibility with any extant labels.)
+
+Furthermore, in keeping with Kubernetes
+[convention](https://kubernetes.io/docs/reference/labels-annotations-taints/#nodekubernetesioinstance-type),
+we should use `node.k8s.aws/` as the prefix for any node labels we
+choose to use.
+
+Lastly, in keeping with another Kubernetes convention (such as
+`kubernetes.io/service-name`), we should use `lower-case-with-hypens`,
+not `camelCase` for words following the `/`.
+
+## Architecture
+
+One paritial solution to the architecture issue brought up above would
+be to remove the ability to specify the `architecture` in the
+provisioner at all. However, this does not seem like a good idea
+because the provisioner is not specific to EC2, and therefore the
+specific behavior of cloud provider behavior (such as launch
+templates) doesn't seem like it merits that kind of change.
+
+Another solution would be to allow the user to specify architecture
+specific labels:
+
+```yaml
+apiVersion: provisioning.karpenter.sh/v1alpha1
+kind: Provisioner
+spec:
+  labels:
+    # applied only to arm64
+    arm64:
+      node.k8s.aws/launch-template-id: id-of-arm64-based-ami
+    x86_64:
+      node.k8s.aws/launch-template-id: id-of-x86_64-based-ami
+    # applied everywhere
+    other-label: other-value
+```
+
+Or:
+
+```yaml
+apiVersion: provisioning.karpenter.sh/v1alpha1
+kind: Provisioner
+spec:
+  # applied everywhere
+  labels:
+     other-label: other-value
+  # applied only to arm64
+  arm64-labels:
+     node.k8s.aws/launch-template-id: id-of-arm64-based-ami
+  # applied only to x86_64
+  x86_64-labels:
+```
+
+Another possibility is to add another "path element" to the label name
+to make it architecture specific:
+
+```yaml
+apiVersion: provisioning.karpenter.sh/v1alpha1
+kind: Provisioner
+spec:
+  labels:
+      node.k8s.aws/launch-template-id/arm64: id-of-arm64-based-ami
+      # or?
+      node.k8s.aws/arm64/launch-template-id: id-of-arm64-based-ami
+```
+
+This might be very non-standard however and defy expectations.
+
+## Recommendation
+
+For now the recommendation is to support the following in provisioner
+`spec.labels`:
+
+- `node.k8s.aws/launch-template-id`: id of launch template
+- `node.k8s.aws/launch-template-version`: version number or `$LATEST`
```

Comment:
In this case, we will not know the architecture of the AMI user has provided in template (unless, there is a way to go check the template > AMI > architecture). Now, if a pod has a different architecture specified in the selector we will not fail and request for a node with this template ID and that will fail without Karpenter knowing about it.

IMO, supporting this might not be much different than supporting 3rd proposal with these labels `node.k8s.aws/launch-template-id/arm64: id-of-arm64-based-ami`. In the latter, we get the name and architecture and use the correct template based on node selector(or default to amd64)?

Reply:
I think this might be now addressed (based on Ellis' suggestion to not allow both `architecture` and these labels at the same time?

Reply:
Are we going to make extra EC2 calls to get the architecture supported in Launch template ID provided in the label?

Also, is there a reason why we are not using this label key - `node.k8s.aws/launch-template-id/arm64:` ?

Reply:
No, the change would be to not try to validate the architecture at all, just simply not allow both `architecture` and launch template `labels` to be specified at the same time in a Provisioner spec.

Reply:
Ok, hopefully addressed the above in latest revision @prateekgogia